### DISCUSSION TITLE
Cross linking between multisites

### DISCUSSION THREAD

**Discussion by carinameissl** [craftcms/cms#9759] [Q&A]
Hey, 
I have a multi site and want to cross-link between these sites. Is there a way to do so? 
(already tried the advanced settings of entry fields - show site menu and manage relations on per site basis, unfortunately neither of these options seem to solve the issue)

Thanks! Carina

**Comment by brandonkelly** [comment]
Relations don’t keep track of which site was selected when the relation was created, so this may not be possible out of the box.

You could maybe keep track of which specific site you want to link to from a a separate field, such as a dropdown using [LJ Dynamic Fields](https://plugins.craftcms.com/craft-dynamic-fields) to dynamically populate the options based on the available sites.

```twig
{{ craft.app.sites.allSites|map(s => {
  value: site.id,
  label: site.name,
})|json_encode }}
```

Then in your templates, you could inject the site value into the relation query:

```twig
{% set relQuery = clone(entry.myEntriesField) %}
{% if entry.relSiteId %}
    {% do relQuery.siteId(entry.relSiteId) %}
{% endif %}

{% for entry in relQuery.all() %}
    ...
{% endfor %}
```

**Reply by aloco** [reply]
Would you consider adding this to the core (option for storing the site with the relation)? For me it´s the first time running into some kind of limitation with multisites

**Reply by brandonkelly** [reply]
I flirted with the idea a bit when thinking through #3584, but it ended up not being necessary to solve that particular issue. It’s still worth considering for this one though…

**Reply by aloco** [reply]
Just read through this as well.

For me the current behaviour of the CP is a little misleading:

I have both options enabled -> show the site menu and manage relations on a per-site basis

1.) create an entry and enable it only for site A 
2.) switch to site B -> when picking an entry you can switch to entries from site A (thanks to show the site menu option)
3.) before saving, the selected entry from site A is marked as enabled within the entry (site B) I am currently editing
4.) after saving, it seems like it is "replacing" that selected entry from site A with the disabled entry from site B, it appears as disabled in the entry relation field.
5.) in templates, I always get the the current site´s base url, not the base url the entry was selected from.

I am not sure what effect the "manage relations on a per site basis option" has

Is there anything I am missing here?

**Reply by brandonkelly** [reply]
@aloco It sounds like the entry you selected is saved to both Site A and Site B; it just happens to be enabled in Site A, and disabled in Site B.

If you don’t want the entry to exist at all in Site B, first make sure its section’s Propagation Method is set to “Let each entry choose which sites it should be saved to”, and then edit the entry in Site B, and choose the “Delete for site” action in the Save button menu.
